File: SafeAround.Api/Persistence/Models/Incident.cs
namespace SafeAround.Api.Persistence.Models;

public class Incident : BaseTrackingEntity
{
    public required string Title { get; set; }
    public required string Description { get; set; }
    public required float Latitude { get; set; }
    public required float Longitude { get; set; }
    public required DateTime OccurrenceDate { get; set; }
    public required Guid UserId { get; set; }
    public AppUser User { get; set; }
}

File: SafeAround.Client/app/src/main/java/com/example/safearound/models/Incident.kt
package com.example.safearound.models

import kotlinx.datetime.Instant
import kotlinx.datetime.LocalDateTime
import kotlinx.serialization.Serializable


@Serializable
data class Incident (
    val id: Int,
    val title: String,
    val description: String,
    val latitude: Double,
    val longitude: Double,
    val categoryId: Int,
    val categoryName: String,
    val categoryCode: String,
    val occurrenceDate: LocalDateTime,
    val userId: String,
){
}